pkg/volume/empty_dir/empty_dir_test.go

```diff
@@ -274,3 +274,42 @@ func TestPluginBackCompat(t *testing.T) {
 		t.Errorf("Got unexpected path: %s", volPath)
 	}
 }
+
+// TestMetrics tests that MetricProvider methods return sane values.
+func TestMetrics(t *testing.T) {
+	// Create an empty temp directory for the volume
+	tmpDir, err := ioutil.TempDir(os.TempDir(), "empty_dir_test")
+	if err != nil {
+		t.Fatalf("Can't make a tmp dir: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	plug := makePluginUnderTest(t, "kubernetes.io/empty-dir", tmpDir)
+
+	spec := &api.Volume{
+		Name: "vol1",
+	}
+	pod := &api.Pod{ObjectMeta: api.ObjectMeta{UID: types.UID("poduid")}}
+	builder, err := plug.NewBuilder(volume.NewSpecFromVolume(spec), pod, volume.VolumeOptions{RootContext: ""})
+	if err != nil {
+		t.Errorf("Failed to make a new Builder: %v", err)
+	}
+
+	// Need to create the subdirectory
+	os.MkdirAll(builder.GetPath(), 0755)
+
+	// TODO(pwittroc): Move this into a reusable testing utility
+	metrics, err := builder.GetMetrics()
+	if err != nil {
+		t.Errorf("Unexpected error when calling GetMetrics %v", err)
+	}
+	if metrics.Used.Value() != 4096 {
```

Comment:
this is very fs type dependent. In tmpfs, directory size is 0, and thus we'd expect != 4096 is correct. A fs has block size more than 4096, see [`stat(2)` ](http://linux.die.net/man/2/stat)